rpc/src/reveal.rs

```diff
@@ -0,0 +1,144 @@
+// RGB node providing smart contracts functionality for Bitcoin & Lightning.
+//
+// Written in 2022 by
+//     Dr. Maxim Orlovsky <orlovsky@lnp-bp.org>
+//
+// Copyright (C) 2022 by LNP/BP Standards Association, Switzerland.
+//
+// You should have received a copy of the MIT License along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+use bitcoin::OutPoint;
+use bp::seals::txout::CloseMethod;
+
+#[derive(From, PartialEq, Eq, StrictEncode, StrictDecode)]
+pub struct Reveal {
+    /// Outpoint blinding factor (generated when the utxo blinded was created)
+    pub blinding_factor: u64,
+
+    /// Locally-controlled outpoint (specified when the utxo blinded was created)
+    pub outpoint: OutPoint,
+
+    /// method (specified when the utxo blinded was created)
+    pub close_method: CloseMethod,
+}
+
+impl std::fmt::Display for Reveal {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}@{}#{}", self.close_method, self.outpoint, self.blinding_factor)
+    }
+}
+
+impl std::fmt::Debug for Reveal {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Reveal")
+            .field("blinding_factor", &self.blinding_factor)
+            .field("outpoint", &self.outpoint)
+            .finish()
+    }
+}
+
+impl Clone for Reveal {
+    fn clone(&self) -> Self {
+        Self {
+            blinding_factor: self.blinding_factor.clone(),
+            outpoint: self.outpoint.clone(),
+            close_method: self.close_method.clone(),
+        }
+    }
+}
```

Comment:
I can understand the custom Display impl, but can't you just derive Clone and Debug on the `Reveal` struct?